src/components/MessageList/MessageList.js

```diff
@@ -39,664 +45,367 @@ const ErrorNotification = styled.View`
-      return <DateSeparator message={message} />;
+      const DateSeparatorComponent = dateSeparator || DateSeparator;
+      return <DateSeparatorComponent message={message} />;
     } else if (message.type === 'channel.event') {
-      const EventIndicator =
-        this.props.eventIndicator || this.props.EventIndicator;
-      return <EventIndicator event={message.event} />;
+      const EventIndicatorComponent = eventIndicator || EventIndicator;
+      return <EventIndicatorComponent event={message.event} />;
     } else if (message.type === 'system') {
       return <MessageSystem message={message} />;
     } else if (message.type !== 'message.read') {
-      const readBy = this.readData[message.id] || [];
       return (
-        <Message
-          client={this.props.client}
-          channel={this.props.channel}
-          onThreadSelect={this.props.onThreadSelect}
+        <DefaultMessage
+          client={client}
+          channel={channel}
+          onThreadSelect={onThreadSelect}
           message={message}
-          groupStyles={groupStyles}
-          Message={this.props.Message}
-          Attachment={this.props.Attachment}
-          readBy={readBy}
-          disabled={this.props.disabled}
-          lastReceivedId={
-            this.state.lastReceivedId === message.id
-              ? this.state.lastReceivedId
-              : null
-          }
-          onMessageTouch={this.props.onMessageTouch}
-          dismissKeyboardOnMessageTouch={
-            this.props.dismissKeyboardOnMessageTouch
-          }
-          setEditingState={this.props.setEditingState}
-          editing={this.props.editing}
-          threadList={this.props.threadList}
-          messageActions={this.props.messageActions}
-          updateMessage={this.props.updateMessage}
-          removeMessage={this.props.removeMessage}
-          retrySendMessage={this.props.retrySendMessage}
-          openThread={this.props.openThread}
-          emojiData={this.props.emojiData}
-          actionSheetStyles={this.props.actionSheetStyles}
-          AttachmentFileIcon={this.props.AttachmentFileIcon}
+          groupStyles={message.groupStyles}
+          Message={Message}
+          Attachment={Attachment}
+          readBy={message.readBy}
+          disabled={disabled}
+          lastReceivedId={lastReceivedId === message.id ? lastReceivedId : null}
+          onMessageTouch={onMessageTouch}
+          dismissKeyboardOnMessageTouch={dismissKeyboardOnMessageTouch}
+          setEditingState={setEditingState}
+          editing={editing}
+          threadList={threadList}
+          messageActions={messageActions}
+          updateMessage={updateMessage}
+          removeMessage={removeMessage}
+          retrySendMessage={retrySendMessage}
+          openThread={openThread}
+          emojiData={emojiData}
+          actionSheetStyles={actionSheetStyles}
+          AttachmentFileIcon={AttachmentFileIcon}
```

Comment:
can we enable the linter abc option? I think alphabetizing things is best practice so searching for what you're looking for is more natural